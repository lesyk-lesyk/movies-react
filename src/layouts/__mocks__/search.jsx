import React from 'react';

export const SearchLayout = () => React.createElement(
  'SearchLayout'
);
