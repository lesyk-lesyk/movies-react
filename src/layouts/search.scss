.search-container {
  padding: 40px 100px;
  max-width: 1000px;
  position: relative;
  margin: 0 auto;
  padding-top: 30px;

  .search-header {
    margin-top: 50px;
    margin-bottom: 25px;
    color: white;
    font-size: 26px;
    font-weight: bold;
    text-transform: uppercase;
  }

  .search-controls {
    margin-top: 25px;
  }
}
