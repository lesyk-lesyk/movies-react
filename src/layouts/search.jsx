import * as React from "react";
import { Header } from '../components/header';
import { SiteName } from '../components/site-name';
import { SearchField } from '../components/search-field';
import { SearchControls } from '../components/search-controls';
import { SubHeader } from '../components/sub-header';
import { MovieList } from '../components/movie-list'
import './search.scss';

export class SearchLayout extends React.Component {
  constructor() {
    super();
    this.movies = [
      {
        poster_path: 'https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg',
        title: 'No Country for Old Men',
        release_date: '2007',
        genres: [
          'Crime',
          'Drama',
          'Thriller'
        ],
      },
      {
        poster_path: 'https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg',
        title: 'No Country for Old Men',
        release_date: '2007',
        genres: [
          'Crime',
          'Drama',
          'Thriller'
        ],
      },
      {
        poster_path: 'https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg',
        title: 'No Country for Old Men',
        release_date: '2007',
        genres: [
          'Crime',
          'Drama',
          'Thriller'
        ],
      },
      {
        poster_path: 'https://image.tmdb.org/t/p/w500/6o0UWX2naW7HK45PDNYmoMIk5qs.jpg',
        title: 'No Country for Old Men',
        release_date: '2007',
        genres: [
          'Crime',
          'Drama',
          'Thriller'
        ],
      }
    ];

    // this.movies = [];
  }
  render() {
    return (
      <React.Fragment>
        <Header>
          <div className="container search-container">
            <SiteName />
            <h1 className="search-header">Find your movie</h1>
            <SearchField />
            <SearchControls />
          </div>
        </Header>
        <SubHeader>
          SubHeader
        </SubHeader>
        <MovieList movies={this.movies} />
      </React.Fragment>
    );
  }
}
