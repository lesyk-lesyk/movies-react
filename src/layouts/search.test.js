import React from 'react';
import { SearchLayout } from './search';
import renderer from 'react-test-renderer';

jest.mock('./search.scss', () => ({}));
jest.mock('../components/header/header');
jest.mock('../components/content/content');
jest.mock('../components/footer/footer');
jest.mock('../components/search-block/search-block');
jest.mock('../components/sub-header/sub-header');
jest.mock('../components/search-counter/search-counter');
jest.mock('../components/sort-controls/sort-controls');
jest.mock('../components/movie-list/movie-list');
jest.mock('../components/error-boundary/error-boundary');

describe('SearchLayout', () => {
  it('renders', () => {
    const tree = renderer
      .create(<SearchLayout />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
