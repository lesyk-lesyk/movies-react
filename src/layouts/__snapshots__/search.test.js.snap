// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchLayout renders 1`] = `
Array [
  <Header>
    <SearchBlock />
    <SubHeader>
      <SearchCounter
        count={4}
      />
      <SortControls />
    </SubHeader>
  </Header>,
  <Content>
    <ErrorBoundary>
      <MovieList />
    </ErrorBoundary>
  </Content>,
  <Footer />,
]
`;
